### ISSUE TITLE
scheme "" not registered, fallback to default scheme

### ISSUE THREAD

**Issue by IanVaughan** [DataDog/heroku-buildpack-datadog#236]
When the buildpack runs, it has the following output:
```
I0122 12:14:10.397689      54 clientconn.go:106] parsed scheme: ""
I0122 12:14:10.397732      54 clientconn.go:106] scheme "" not registered, fallback to default scheme
I0122 12:14:10.397802      54 passthrough.go:48] ccResolverWrapper: sending update to cc: {[{localhost:5001  <nil> 0 <nil>}] <nil> <nil>}
I0122 12:14:10.406986      54 clientconn.go:933] ClientConn switching balancer to "pick_first"
I0122 12:14:10.407120      54 clientconn.go:106] parsed scheme: ""
I0122 12:14:10.407144      54 clientconn.go:106] scheme "" not registered, fallback to default scheme
I0122 12:14:10.407186      54 passthrough.go:48] ccResolverWrapper: sending update to cc: {[{localhost:5001  <nil> 0 <nil>}] <nil> <nil>}
I0122 12:14:10.407206      54 clientconn.go:933] ClientConn switching balancer to "pick_first"
```

```
===============
Agent (v7.25.0)
===============

  Status date: 2021-01-22 12:29:08.890285 UTC
  Agent start: 2021-01-22 12:27:40.464750 UTC
  Pid: 56
  Go Version: go1.14.12
  Python Version: 3.8.5
  Build arch: amd64
  Agent flavor: agent
  Check Runners: 4
  Log File: /app/.apt/var/log/datadog/datadog.log
  Log Level: ERROR

  Paths
  =====
    Config File: /app/.apt/etc/datadog-agent/datadog.yaml
    conf.d: /app/.apt/etc/datadog-agent/conf.d
    checks.d: /app/.apt/etc/datadog-agent/checks.d

  Clocks
  ======
    NTP offset: -2.369ms
    System UTC time: 2021-01-22 12:29:08.890285 UTC

  Host Info
  =========
    bootTime: 2021-01-14 20:04:57.000000 UTC
    kernelArch: x86_64
    kernelVersion: 4.4.0-1083-aws
    os: linux
    platform: ubuntu
    platformFamily: debian
    platformVersion: 20.04
    procs: 4
    uptime: 184h22m48s
    virtualizationRole: guest
    virtualizationSystem: lxc

  Hostnames
  =========
    hostname: out-fund-staging.run.2445
    socket-fqdn: e5b0b4f8-da6e-481c-b088-f581513f36a5.prvt.dyno.rt.heroku.com.
    socket-hostname: e5b0b4f8-da6e-481c-b088-f581513f36a5
    host tags:
      dyno:run.2445
      dynotype:run
      buildpackversion:dev
      appname:out-fund-staging
      enviroment:staging
    hostname provider: configuration

  Metadata
  ========
    hostname_source: configuration

=========
Collector
=========

  Running Checks
  ==============

    cpu
    ---
      Instance ID: cpu [OK]
      Configuration Source: file:/app/.apt/etc/datadog-agent/conf.d/cpu.d/conf.yaml.default
      Total Runs: 6
      Metric Samples: Last Run: 8, Total: 42
      Events: Last Run: 0, Total: 0
      Service Checks: Last Run: 0, Total: 0
      Average Execution Time : 0s
      Last Execution Date : 2021-01-22 12:29:00.000000 UTC
      Last Successful Execution Date : 2021-01-22 12:29:00.000000 UTC


    disk (4.0.0)
    ------------
      Instance ID: disk:e5dffb8bef24336f [OK]
      Configuration Source: file:/app/.apt/etc/datadog-agent/conf.d/disk.d/conf.yaml.default
      Total Runs: 6
      Metric Samples: Last Run: 220, Total: 1,320
      Events: Last Run: 0, Total: 0
      Service Checks: Last Run: 0, Total: 0
      Average Execution Time : 50ms
      Last Execution Date : 2021-01-22 12:29:07.000000 UTC
      Last Successful Execution Date : 2021-01-22 12:29:07.000000 UTC


    file_handle
    -----------
      Instance ID: file_handle [OK]
      Configuration Source: file:/app/.apt/etc/datadog-agent/conf.d/file_handle.d/conf.yaml.default
      Total Runs: 5
      Metric Samples: Last Run: 5, Total: 25
      Events: Last Run: 0, Total: 0
      Service Checks: Last Run: 0, Total: 0
      Average Execution Time : 0s
      Last Execution Date : 2021-01-22 12:28:59.000000 UTC
      Last Successful Execution Date : 2021-01-22 12:28:59.000000 UTC


    io
    --
      Instance ID: io [OK]
      Configuration Source: file:/app/.apt/etc/datadog-agent/conf.d/io.d/conf.yaml.default
      Total Runs: 6
      Metric Samples: Last Run: 182, Total: 966
      Events: Last Run: 0, Total: 0
      Service Checks: Last Run: 0, Total: 0
      Average Execution Time : 3ms
      Last Execution Date : 2021-01-22 12:29:06.000000 UTC
      Last Successful Execution Date : 2021-01-22 12:29:06.000000 UTC


    load
    ----
      Instance ID: load [OK]
      Configuration Source: file:/app/.apt/etc/datadog-agent/conf.d/load.d/conf.yaml.default
      Total Runs: 5
      Metric Samples: Last Run: 6, Total: 30
      Events: Last Run: 0, Total: 0
      Service Checks: Last Run: 0, Total: 0
      Average Execution Time : 0s
      Last Execution Date : 2021-01-22 12:28:58.000000 UTC
      Last Successful Execution Date : 2021-01-22 12:28:58.000000 UTC


    memory
    ------
      Instance ID: memory [OK]
      Configuration Source: file:/app/.apt/etc/datadog-agent/conf.d/memory.d/conf.yaml.default
      Total Runs: 6
      Metric Samples: Last Run: 18, Total: 108
      Events: Last Run: 0, Total: 0
      Service Checks: Last Run: 0, Total: 0
      Average Execution Time : 0s
      Last Execution Date : 2021-01-22 12:29:05.000000 UTC
      Last Successful Execution Date : 2021-01-22 12:29:05.000000 UTC


    network (2.0.0)
    ---------------
      Instance ID: network:5c571333f400457d [OK]
      Configuration Source: file:/app/.apt/etc/datadog-agent/conf.d/network.d/conf.yaml.default
      Total Runs: 5
      Metric Samples: Last Run: 31, Total: 155
      Events: Last Run: 0, Total: 0
      Service Checks: Last Run: 0, Total: 0
      Average Execution Time : 7ms
      Last Execution Date : 2021-01-22 12:28:57.000000 UTC
      Last Successful Execution Date : 2021-01-22 12:28:57.000000 UTC


    ntp
    ---
      Instance ID: ntp:d884b5186b651429 [OK]
      Configuration Source: file:/app/.apt/etc/datadog-agent/conf.d/ntp.d/conf.yaml.default
      Total Runs: 1
      Metric Samples: Last Run: 1, Total: 1
      Events: Last Run: 0, Total: 0
      Service Checks: Last Run: 1, Total: 1
      Average Execution Time : 375ms
      Last Execution Date : 2021-01-22 12:27:45.000000 UTC
      Last Successful Execution Date : 2021-01-22 12:27:45.000000 UTC


    uptime
    ------
      Instance ID: uptime [OK]
      Configuration Source: file:/app/.apt/etc/datadog-agent/conf.d/uptime.d/conf.yaml.default
      Total Runs: 6
      Metric Samples: Last Run: 1, Total: 6
      Events: Last Run: 0, Total: 0
      Service Checks: Last Run: 0, Total: 0
      Average Execution Time : 0s
      Last Execution Date : 2021-01-22 12:29:04.000000 UTC
      Last Successful Execution Date : 2021-01-22 12:29:04.000000 UTC

========
JMXFetch
========

  Information
  ==================
  Initialized checks
  ==================
    no checks

  Failed checks
  =============
    no checks

=========
Forwarder
=========

  Transactions
  ============
    Deployments: 0
    Dropped: 0
    DroppedOnInput: 0
    Nodes: 0
    Pods: 0
    ReplicaSets: 0
    Requeued: 0
    Retried: 0
    RetryQueueSize: 0
    Services: 0

  Transaction Successes
  =====================
    Total number: 12
    Successes By Endpoint:
      check_run_v1: 5
      intake: 2
      series_v1: 5

  API Keys status
  ===============
    API key ending with fb296: API Key valid

==========
Endpoints
==========
  https://app.datadoghq.eu - API Key ending with:
      - fb296

==========
Logs Agent
==========


  Logs Agent is not running

=========
APM Agent
=========
  Status: Running
  Pid: 58
  Uptime: 88 seconds
  Mem alloc: 15,978,888 bytes
  Hostname: out-fund-staging.run.2445
  Receiver: localhost:8126
  Endpoints:
    https://trace.agent.datadoghq.eu

  Receiver (previous minute)
  ==========================
    No traces received in the previous minute.
    Default priority sampling rate: 100.0%

  Writer (previous minute)
  ========================
    Traces: 0 payloads, 0 traces, 0 events, 0 bytes
    Stats: 0 payloads, 0 stats buckets, 0 bytes

=========
Aggregator
=========
  Checks Metric Sample: 2,745
  Dogstatsd Metric Sample: 467
  Event: 1
  Events Flushed: 1
  Number Of Flushes: 5
  Series Flushed: 1,676
  Service Check: 47
  Service Checks Flushed: 44

=========
DogStatsD
=========
  Event Packets: 0
  Event Parse Errors: 0
  Metric Packets: 466
  Metric Parse Errors: 0
  Service Check Packets: 0
  Service Check Parse Errors: 0
  Udp Bytes: 35,269
  Udp Packet Reading Errors: 0
  Udp Packets: 342
  Uds Bytes: 0
  Uds Origin Detection Errors: 0
  Uds Packet Reading Errors: 0
  Uds Packets: 0
```

**Comment by yboulkaid** [comment]
+1, we've been running into this too. Let me know if you need more information to debug this :)

**Comment by arapulido** [comment]
This comes from the agent, not the buildpack. I will report it to the agent team. This started on agent `7.24`

**Comment by januszm** [comment]
Happens on AWS Elastic Beanstalk (not Heroku, I know ... same error though), version `7.25.1`

**Comment by arapulido** [comment]
Yes, this is related to the agent, not the buildpack. I am following this up with the agent team. Thanks!

**Comment by suaron** [comment]
I found a workaround for this issue.

```
heroku config:set DD_AGENT_VERSION=7.23.1-1
```

**Comment by arapulido** [comment]
> I found a workaround for this issue.
> 
> ```
> heroku config:set DD_AGENT_VERSION=7.23.1-1
> ```

We don't recommend downgrading the agent. The log messages are harmless and will be removed in the next agent version. We recommend staying in the latest version of the agent and ignoring that output. It will be fixed in the next version.

**Comment by jeremywadsack** [comment]
> The log messages are harmless 

They are confusing to the people who see them. :) I've set `DD_LOG_LEVEL=CRITICAL` to [reduce noise when launching a console or running a command](https://github.com/DataDog/heroku-buildpack-datadog/issues/23) but these still come up.  That implies that this is a critical problem and I should be doing something about it. 


> will be removed in the next agent version

I just set up DataDog this week with the latest version and they are still showing up:

```
===============
Agent (v7.26.0)
===============

  Status date: 2021-04-02 19:06:51.058279 UTC
  Agent start: 2021-04-02 19:05:57.840238 UTC
  Pid: 59
  Go Version: go1.14.12
  Python Version: 3.8.5
  Build arch: amd64
  Agent flavor: agent
  Check Runners: 4
  Log File: /app/.apt/var/log/datadog/datadog.log
  Log Level: CRITICAL
```

It there an issue in DataDog/datadog-agent linked to this? I couldn't find one and didn't see anything in the CHANGELOG related to it.

**Comment by arapulido** [comment]
This is now fixed in agents 6.27 and 7.27
